sys/sleepq.c

```diff
@@ -193,7 +193,20 @@ static void sq_leave(thread_t *td, sleepq_chain_t *sc, sleepq_t *sq) {
   }
 }
 
+static uint32_t tdf_of_slpf(sleep_flags_t flags) {
+  uint32_t td_flags = 0;
+  if (flags & SLPF_INT)
+    td_flags |= TDF_SLEEP_INT;
+  return td_flags;
+}
+
 void sleepq_wait(void *wchan, const void *waitpt) {
+  slp_wakeup_t r = sleepq_wait_abortable(wchan, waitpt, 0);
+  assert(r == SLEEPQ_WKP_REG);
+}
+
+slp_wakeup_t sleepq_wait_abortable(void *wchan, const void *waitpt,
+                                   sleep_flags_t f) {
```

Comment:
1) I suppose so but I couldn't think of anything better. In the end when you want non-abortable sleep you'll probably use `sleepq_wait`
2) In my mind `interruptible` is a subtype of `abortable`. Another subtype of `abortable` would be (not implemented yet) `timed`. So in the context of `signal` we're talking about `interruptible` sleep (which is also `abortable`) because signals specifically use `SLEEPQ_WKP_INT`. We can still work on the terminology but I think this one makes some sense.

Reply:
1. Now it makes as much sense as function `sleepq_broadcast` which would do signal or broadcast depending on flag argument.
2. Ok. :wink:

Reply:
1. I'll leave `abortable` unless I get a better idea or somebody gives me a better idea.

Reply:
Better idea: `sleepq_wait`, `sleepq_wait_sig`, `sleepq_timedwait`, `sleepq_timedwait_sig`. :wink:

Reply:
Now add "killable" :)